### ISSUE TITLE
osd_timeout should be calculated instead of a fixed time

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1260]
After noticing twice on customer environments that `osd_timeouts` occur to much in the `alba-proxy` we've decided to create this ticket. 

Currently the default is 2 seconds before a osd in a certain proxy is put into timeout. After discussion with @domsj and after testing this on customer environments we've agreed to increase this to 5 seconds.

This should be done on:
* The proxies used for the volumedriver
* The proxies used for scrubbing

**Comment by wimpers** [comment]
@kinvaris @domsj  why is this something for the Framework? Why don't we update the default directly instead of setting 'a better default through the framework'?

**Comment by kinvaris** [comment]
@wimpers because the framework deploys the configuration, this is easily changed and repackaged. As if we would have to change it in alba a recompile would need to take place.

**Comment by wimpers** [comment]
I disagree, it is up to alba to set a sane default and not upto the framework to patch every environment. Less chance for errors.

**Comment by toolslive** [comment]
2.0 seconds is a perfectly sane value. whether it's enough or even a good choice depends on other things like for example the fragment size in the preset, or the setup (compare geo to all-flash). 
Alba cannot know what you have there.

Mind, It's no issue setting this to 5.0s, but it's an illusion to think that 5.0s is perfect/good/sane/goodenough for every deployment.

**Comment by wimpers** [comment]
@toolslive I believe the proxy default was updated in https://github.com/openvstorage/alba/commit/0d0c4ab right?

I agree with you a better strategy than a default is to set the time-out based upon fragment size or other (to be defined) parameters. In that case it makes sense to 'calculate' and have the timeout being set by the framework. I'll keep this ticket open for that 'calculated' timeout.

**Comment by toolslive** [comment]
No. the relevant value is this:
https://github.com/openvstorage/alba/blob/master/ocaml/src/cli_proxy.ml#L39
which is still 2.0s

**Comment by wimpers** [comment]
@domsj @toolslive do we keep the default now at 2 seconds in ALBA or not?

**Comment by wimpers** [comment]
Default has been changed to 10s (https://github.com/openvstorage/alba/pull/557)

**Comment by JeffreyDevloo** [comment]
@wimpers 
Do you think this ticket has been resolved with upping the timeout?

**Comment by wimpers** [comment]
No as he the osd-timeout is fixed, making it dynamic is why this ticket is open.
